### PR TITLE
Add new movement mode - jogging

### PR DESCRIPTION
**Pull request by Demoragon**
#### Summary

SUMMARY: Features "Split run into run and sprint"

#### Purpose of change
Allow player to move faster but not as fast as running. 

#### Describe the solution
~~Add new movement mode and write some code, which is very similar to other movement type's code.~~
Rename run as sprint, add new movement mode (new run). 
Overhaul monster speeds (haven't done yet)

#### Describe alternatives you've considered
~~Probably, different values for speed and stamina multipliers would be better, but I'm not sure.~~

#### Testing
~~Start new game, use jogging, stop jogging by using action menu, set a key for jogging in keybindings menu and then used it. While jogging character moves faster and gets tired.~~

### CODE REVIEW

File: data/json/move_modes.json [hunk 1]
```diff
@@ -52,5 +52,27 @@
...
+    "panel_color": "yellow",
+    "symbol_color": "yellow",
+    "exertion_level": "BRISK_EXERCISE",
+    "change_good_none": "You start jogging.",
+    "change_good_animal": "You spur your steed to go faster.",
+    "change_good_mech": "You increase the power of your mech's leg servos.",
+    "change_bad_none": "You're too tired to jog.",
+    "change_bad_animal": "Your steed is too tired to go faster.",
+    "change_bad_mech": "Your mech's leg servos are unable to operate faster.",
+    "move_type": "jogging",
```
Comment (anothersimulacrum) [inline data/json/move_modes.json:71]: I don't think this needs a new move mode type, jogging fits perfectly well as a `running` move_mode_type.
```suggestion
    "move_type": "running",
```

File: src/character.cpp [hunk 2]
```diff
@@ -1710,6 +1710,11 @@ bool Character::is_running() const
     return move_mode->type() == move_mode_type::RUNNING;
 }
 
+bool Character::is_jogging() const
+{
+    return move_mode->type() == move_mode_type::JOGGING;
```
Comment (anothersimulacrum) [inline src/character.cpp:1715]:
```suggestion
    return move_mode == move_mode_id( "jogging" );
```

File: src/move_mode.cpp [hunk 3]
```diff
@@ -31,6 +31,7 @@ const move_mode &move_mode_id::obj() const
 static const std::map<std::string, move_mode_type> move_types {
     { "crouching", move_mode_type::CROUCHING },
     { "walking",   move_mode_type::WALKING },
+    { "jogging",   move_mode_type::JOGGING },
```
Comment (anothersimulacrum) [inline src/move_mode.cpp:34]:
```suggestion
```

File: src/move_mode.h [hunk 4]
```diff
@@ -25,6 +25,7 @@ enum class steed_type : int {
 enum class move_mode_type : int {
     CROUCHING,
     WALKING,
+    JOGGING,
```
Comment (anothersimulacrum) [inline src/move_mode.h:28]:
```suggestion
```

File: data/json/move_modes.json [hunk 5]
```diff
@@ -52,5 +52,27 @@
     "sound_multiplier": 0.5,
     "move_speed_multiplier": 0.5,
     "swim_speed_mod": 50
+  },
+  {
+    "type": "movement_mode",
+    "id": "jogging",
```
Comment (anothersimulacrum) [inline data/json/move_modes.json:58]: Also, could I suggest changing this to `"jog"` to fit with the other types?

File: src/character.cpp (hunk 2 above)
Comment (Demoragon) [reply src/character.cpp]: Should I also change Character::is_running?

**Comment by dissociativity** [comment]
I mean, we do need a movement mode that won't burn you out heavily by running what feels to be about 50 meters with a relatively light load.

**Comment by KorGgenT** [comment]
> and overall the standard monster speed (your standard zombies as reference) to be faster than a walk, but slower than a run

No, this is the opposite of what kevin said.

**Comment by dissociativity** [comment]
my bad, I'm sure the clarification from you is useful though.

File: src/character.cpp (hunk 2 above)
Comment (anothersimulacrum) [reply src/character.cpp]: I missed this, and it's some irrelevant now, but I think renaming this whole trio is_running/walking/crouching and the enum associated might be a good idea.

**Comment by Demoragon** [comment]
> The biggest thing here is you have no rationale for what you're trying to add, in some cases it's obvious, but in this case there are many hypothetical reasons to make the change, and it's impossible to evaluate whether you're accomplishing your goal because you aren't saying what it is.

Sorry for that. My goal is to add new movement mode that quite fits your definition of running. Something between regular walking and moving as fast as possible. 
> That's kind of irrelevant though because a lot of thought and discussion has gone into this specific thing.
> It was very intentional that we don't have a gait at this speed.

Sorry again.
> What I've outlined previously as being an option is three gaits:
> Walk is your top speed that doesn't burn stamina, ideally it slowly recovers it.
> Running is significantly faster and burns more energy, but is only mildly stamina negative, as in you can keep it up over a reasonable duration unless you're heavily encumbered.
> Sprinting is roughly where running is right now, move as fast as possible at the cost of rapidly depleting stamina, bottoming it out after 100 tiles or so.

So I guess, I should change some values (make jogging faster and burning more stamina) and rename "jogging" as "running" and "running" as "sprinting"?
> In tandem with this adjustment, you _must_ overhaul monster speeds relative to player speeds such that a large number of enemies fall between running and sprinting. 

Should I do it in this PR or in the next one?

**Comment by kevingranade** [comment]
The gait change and the monster speed change definitely need to happen at the same time.
It's possible that this could be accomplished with an adjustment to "moves per tile for all monsters" instead of editing every monster seperately, so it might be a smaller change than you might expect.

**Pull request closed without merging**